File: 1135-connecting-cities-with-minimum-cost/1135-connecting-cities-with-minimum-cost.cpp
class Solution {
public:
    int minimumCost(int n, vector<vector<int>>& connections) {
        vector<int> p(n);
        iota(p.begin(), p.end(), 0);
        sort(connections.begin(), connections.end(), [](auto& a, auto& b) {return a[2]<b[2];});

        function<int(int)> find = [&](int x) {
            if(x != p[x]) p[x] = find(p[x]);
            return p[x];
        };
        int ans = 0;
        for(const auto& con : connections) {
            int x = con[0]-1, y = con[1]-1, cost = con[2];
            if(find(x) == find(y)) continue;
            p[find(x)] = find(y);
            ans += cost;
            if(--n == 1) return ans;
        }
        return -1;
        
    }
};

File: 1135-connecting-cities-with-minimum-cost/1135-connecting-cities-with-minimum-cost.py
class Solution(object):
    def minimumCost(self, n, connections):
        """
        :type n: int
        :type connections: List[List[int]]
        :rtype: int
        """
        def find(x):
            if p[x] != x:
                p[x] = find(p[x])
            return p[x]

        connections.sort(key=lambda x: x[2])
        p = list(range(n))
        ans = 0
        for x, y, cost in connections:
            x, y = x - 1, y - 1
            if find(x) == find(y):
                continue
            p[find(x)] = find(y)
            ans += cost
            n -= 1
            if n == 1:
                return ans
        return -1